### PR TITLE
MDEV-9007: Systemd not to use bootstrap

### PR DESCRIPTION
**Pull request by grooverdan**

### CODE REVIEW

**Comment by svoj** [comment]
Hi Daniel,

thanks for your contribution. JIRA task link (just for the record): https://mariadb.atlassian.net/browse/MDEV-9007
This task was added to 10.1.9 backlog, which is planned to be handled between 2015-10-27 and 2015-11-17.

Thanks,
Sergey

File: support-files/mariadb@.service.in [hunk 1]
```diff
@@ -60,7 +60,6 @@ PermissionsStartOnly=true
 # Use the [service] section and Environment="MYSQLD_OPTS=...".
 # This isn't a replacement for my.cnf.
 
-ExecStart=
```
Comment (svoj) [inline support-files/mariadb@.service.in]: Wasn't this line supposed to reset ExecStart inherited from mariadb.service?

File: scripts/galera_new_cluster.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+
+VERSION="@VERSION@@MYSQL_SERVER_SUFFIX@"
+COMPILATION_COMMENT="@COMPILATION_COMMENT@"
+
+/usr/bin/systemctl set-environment MYSQLD_OPTIONS='--wsrep-new-cluster' && \
+     /usr/bin/systemctl start mariadb.service
+
+/usr/bin/systemctl set-environment MYSQLD_OPTIONS=''
```
Comment (svoj) [inline scripts/galera_new_cluster.sh]: Just some thoughts. Even though the right place to list mysqld options is my.cnf, wasn't MYSQLD_OPTIONS supposed to be used by admins? If so, it's quite unexpected that this script sets and then resets it, without keeping original value.

File: support-files/mysql.server.sh [hunk 3]
```diff
@@ -443,11 +443,8 @@ case "$mode" in
     exit $r
     ;;
   'bootstrap')
-      # Bootstrap the cluster, start the first node
-      # that initiate the cluster
-      echo $echo_n "Bootstrapping the cluster.. "
-      $0 start $other_args --wsrep-new-cluster
-      exit $?
+      log_failure_msg "Please use galera_new_cluster to start the mariadb service with --wsrep-new-cluster"
+      exit 1
```
Comment (svoj) [inline support-files/mysql.server.sh]: I believe we still support 10.1 on pre-systemd distributions, while galera_new_cluster is systemd only thing.

File: support-files/mariadb@.service.in (hunk 1 above)
Comment (grooverdan) [inline support-files/mariadb@.service.in]: mariadb@.service isn't inherited from mariadb.service so this fix is correct.

File: scripts/galera_new_cluster.sh (hunk 2 above)
Comment (grooverdan) [inline scripts/galera_new_cluster.sh]: I was wondering how to do this. There isn't a good systemctl get-environment interface so I'll add a new variable $WSREP_NEW_CLUSTER to the service.in files and use that in the script. I'll also add an optional script argument of the service name defaulting to mariadb.service.

File: support-files/mysql.server.sh (hunk 3 above)
Comment (grooverdan) [inline support-files/mysql.server.sh]: both /lib/lsb/init-functions and /etc/init.d/functions set _use_systemctl = 1 when systemd is available so I'll use that condition here.

**Comment by grooverdan** [comment]
been merged. Closing.

**Pull request closed without merging**